### PR TITLE
Define APIs to access the database

### PR DESCRIPTION
**Pull request by joumanakh**

### CODE REVIEW

**Review by Karyum — COMMENTED** [review]
so @joumanakh we should arrange a zoom meeting with @saherhanif, since the DB structure has changed for the final time.
also this pull request for me looks a bit weird cause you have done like 70% of the API work 😅 [looking at your task](https://github.com/orgs/WebAhead-cs2/projects/2?pane=issue&itemId=18702421) I don't see much details or description so for me personally this is a valid pull request.
there's couple of things we need to discuss and change, instead of a long review, it's better if we have a zoom meeting

File: database/models/site.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    site_id,
+    name,
+    country,
+  ]);
+};
+export const deleteSite = (site_id) => {
+  db.query(`DELETE FROM site WHERE site_id = ($1)`, [site_id]);
+};
+export const getSites = async () => {
+  let siteTable = await db.query(`SELECT * FROM site`);
```
Comment (tawfiknasser) [inline database/models/site.js]: use `const` instead ?

File: database/models/scrum.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const db = require("./database/connections");
+const cors = require("cors");
+app.use(cors());
```
Comment (tawfiknasser) [inline database/models/scrum.js]: Is app defined in this module ? Are you binding it from somewhere ? Why are we using `app` here at all. Data models shall be pure logic to the database.

File: database/models/reports.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const db = require("./database/connections");
+const cors = require("cors");
+app.use(cors());
+export const ProductivityEmployee = async () => {
+  // let ProductivityEmployeeTable = await db.query(`SELECT employee.idnumber,capacity.productivity FROM employee INNER JOIN capacity ON employee.id=capacity.employee_id`);
```
Comment (tawfiknasser) [inline database/models/reports.js]: I wouldn't keep commented code.

File: database/models/reports.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const db = require("./database/connections");
+const cors = require("cors");
+app.use(cors());
+export const ProductivityEmployee = async () => {
```
Comment (tawfiknasser) [inline database/models/reports.js]: Let's consist one style.
Model name start with capital letter.
Model methods are camel case.
All methods shall be wrapped with `async` or return `Promise`

File: database/models/reports.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  // let ProductivityEmployeeTable = await db.query(`SELECT employee.idnumber,capacity.productivity FROM employee INNER JOIN capacity ON employee.id=capacity.employee_id`);
+  let ProductivityEmployeeTable = await db.query(
```
Comment (tawfiknasser) [inline database/models/reports.js]: `const`

File: database/models/scrum.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+export const createScrum = (name, scrum_master, application_id, project_id) => {
+  db.query(
```
Comment (tawfiknasser) [inline database/models/scrum.js]: `createScrum` does not return anything ?
How can we indicate if the creation succeeded ?

File: database/models/reports.js (hunk 4 above)
Comment (tawfiknasser) [reply database/models/reports.js]: And I would wrap the function body with `try{}catch(e){}` block.

File: database/models/project.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  );
+};
+export const editProject = (id, name, start_date, pi_count) => {
+  db.query(
+    `UPDATE project SET name= ($2), start_date=($3), pi_count=($4) WHERE id = ($1)`,
+    [id, name, start_date, pi_count]
+  );
+};
+export const deleteProject = (id) => {
+  db.query(`DELETE FROM project WHERE id = ($1)`, [id]);
```
Comment (tawfiknasser) [inline database/models/project.js]: We shall have no deletion for any kind of data. 
Only archive.

File: server.js [hunk 8]
```diff
@@ -1,26 +1,304 @@
...
+  const { id, name, start_date, pi_count } = req.body;
+  Project.editProject(id, name, start_date, pi_count);
+});
+app.post("/deleteProject", async (req, res) => {
+  const { id } = req.body;
+  Project.deleteProject(id);
+});
+app.get("/getProjects", async (req, res) => {
+  let ProjectTable = Project.getProjects();
+  res.send(JSON.stringify({ Project: ProjectTable }));
```
Comment (tawfiknasser) [inline server.js:280]: The task was about developing the Models. This is out of scope.

**Review by tawfiknasser — CHANGES_REQUESTED** [review]

File: server.js (hunk 8 above)
Comment (tawfiknasser) [reply server.js:280]: This could be a good way to test your code locally. Actually better is writing unit test. It could be also an example how to use the models. In any way we should not touch `server.js` in this task.
